docs/GroupDefs.hpp

```diff
@@ -159,155 +164,485 @@
  * Charm++
 
 SpECTRE builds a layer on top of Charm++ that performs various safety checks and
-initialization for the user that can otherwise lead to difficult to debug
+initialization for the user that can otherwise lead to difficult-to-debug
 undefined behavior. The central concept is what is called a %Parallel
 Component. A %Parallel Component is a struct with several type aliases that
 is used by SpECTRE to set up the Charm++ chares and allowed communication
-patterns. It might be most natural to think of %Parallel Components
-as input arguments to a metaprogram that writes the parallelization
-infrastructure that you requested for the executable. There is no restriction
-on the number of %Parallel Components, though practically it is best to have
-around 10 at most.
-
-Each %Parallel Component must have the following type aliases:
-1. `using chare_type` is set to one of `Parallel::Algorithms::Singleton`,
-   `Parallel::Algorithms::Array`, `Parallel::Algorithms::Group`, or
-   `Parallel::Algorithms::Nodegroup`. What these mean is explained below.
+patterns. %Parallel Components are input arguments to the compiler, which then
+writes the parallelization infrastructure that you requested for the executable.
+There is no restriction on the number of %Parallel Components, though
+practically it is best to have around 10 at most.
```

Comment:
It would be good to give a short overview here of the concepts to be explained in detail below. I suggest a list like:

- Metavariables: Provides high-level configuration to the compiler, e.g. the physical system to be simulated.
- Phase: Defines distinct simulation phases separated by a global synchronisation point, e.g. `Initialize`, `Evolve` and `Exit`.
- Algorithm: In each phase, asks all parallel components to execute their actions.
- Parallel component: Maintains and executes its actions, e.g. repeats a sequence of actions until a condition is met.
- Action: Performs a small computational task, e.g. computing an integral over a small domain. May require input from other actions.